plugin/src/main/java/org/gradle/testretry/internal/config/TestRetryTaskExtensionAdapter.java

```diff
@@ -38,28 +38,22 @@
     private final ProviderFactory providerFactory;
     private final TestRetryTaskExtension extension;
     private final boolean simulateNotRetryableTest;
-    private boolean useConventions;
+    private final boolean useConventions;
 
     public TestRetryTaskExtensionAdapter(
         ProviderFactory providerFactory,
         TestRetryTaskExtension extension,
         VersionNumber gradleVersion
-
     ) {
         this.providerFactory = providerFactory;
         this.extension = extension;
         this.simulateNotRetryableTest = Boolean.getBoolean(SIMULATE_NOT_RETRYABLE_PROPERTY);
+        this.useConventions = gradleVersion.compareTo(VersionNumber.version(5, 1)) >= 0;
 
-        boolean gradle51OrLater = gradleVersion.getMajor() == 5
-            ? gradleVersion.getMinor() >= 1
-            : gradleVersion.getMajor() > 5;
-
-        this.useConventions = gradle51OrLater;
-
-        initialize(extension, gradle51OrLater);
+        initialize(extension, this.useConventions);
     }
 
-    private void initialize(TestRetryTaskExtension extension, boolean gradle51OrLater) {
+    private static void initialize(TestRetryTaskExtension extension, boolean gradle51OrLater) {
```

Comment:
I'd keep this non-static and reference the `useConventions` field directly instead of passing a boolean parameter.

Reply:
That's called from the constructor, so it's important that it doesn't depend on the state yet

Reply:
Ok, fair enough.